File: 7-kyu/strongn-strong-number-(special-numbers-series-#2)-/swift/solution.swift
func strongNumber(_ number: Int) -> String {
    let numberArr = String(number).compactMap { Int(String($0)) }
    let factorialSum = numberArr.map { factorial($0) }.reduce(0, +)
    return number == factorialSum ? "STRONG!!!!" : "Not Strong !!"
}

func factorial(_ number: Int) -> Int {
    guard number > 2 else { return number }
    return (2...number).reduce(1, *)
}

File: 7-kyu/strongn-strong-number-(special-numbers-series-#2)-/javascript/solution.js
const fact = (number) => {
  if (number === 0) {
    return 1;
  } else if (number <= 2) {
    return number;
  }
  let ret = 1;
  for (let i = 2; i <= number; i++) {
    ret *= i;
  }
  return ret;
};

const strong = (n) => {
  const ret = [...(n.toString())].map((elem) => fact(parseInt(elem))).reduce((acc, nxt) => acc + parseInt(nxt), 0);
  return ret === n ? "STRONG!!!!" : "Not Strong !!";
};
